datafusion-examples/examples/dataframe.rs

```diff
@@ -41,3 +44,66 @@ async fn main() -> Result<()> {
 
     Ok(())
 }
+
+// Example to read data from a csv file with inferred schema
+async fn example_read_csv_file_with_inferred_schema() -> Arc<DataFrame> {
+    let path = "example.csv";
+    // Create the data to put into the csv file with headers
+    let content = r#"id,time,vote,unixtime,rating
+    a1,\"10 6, 2013\",3,1381017600,5.0
+    a2,\"08 9, 2013\",2,1376006400,4.5"#;
+    // write the data
+    fs::write(path, content).expect("Problem with writing file!");
```

Comment:
minor nit: this code is repeated a few times and could potentially be moved to its own function

Reply:
Sure thing. As my first commit, I just want to add some stuff. I have a lot more examples to add then I will update it and create a separate function for each type of data files like excel, json etc.

Reply:
I have refactored it according to your suggestion. Thanks